File: algorithm/tree/binary_search_tree/cplus/src/binaryTree.cpp
#include "binaryTree.h"
#include <stdio.h>
#include <vector>

using std::vector;

void BinaryTree::inorderVisit(Node *root) {
    vector<Node *> list;
    while (root != NULL || !list.empty()) {
        while (root != NULL) {
            list.push_back(root);
            root = root->left;
        }
        root = list.at(list.size() - 1);
        list.pop_back();
        printf("%d ", root->val);
        root = root->right;
    }
    printf("\n");
}

File: algorithm/tree/binary_search_tree/java/BinaryTree.java
import java.util.ArrayList;
import java.util.List;

public class BinaryTree {
    public static void inorderVisit(Node root) {
        ArrayList<Node> list = new ArrayList<Node>();
        while (root != null || !list.isEmpty()) {
            while (root != null) {
                list.add(root);
                root = root.left;
            }
            root = list.get(list.size() - 1);
            list.remove(list.size() -1);
            System.out.print(root.val + " ");
            root = root.right;
        }
        System.out.println("");
    }
}
